app/bundles/LeadBundle/Model/CompanyReportData.php

```diff
@@ -0,0 +1,153 @@
+    {
+        $companyColumns = $this->getCompanyColumns();
+        foreach ($companyColumns as $key => $column) {
+            if ($event->hasColumn($key)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return array
+     */
+    private function getCompanyColumns()
+    {
+        return [
+            'comp.id' => [
+                'label' => 'mautic.lead.report.company.company_id',
+                'type'  => 'int',
+                'link'  => 'mautic_company_action',
+            ],
+            'comp.companyname' => [
+                'label' => 'mautic.lead.report.company.company_name',
+                'type'  => 'string',
+                'link'  => 'mautic_company_action',
+            ],
+            'comp.companycity' => [
+                'label' => 'mautic.lead.report.company.company_city',
+                'type'  => 'string',
+                'link'  => 'mautic_company_action',
+            ],
+            'comp.companystate' => [
+                'label' => 'mautic.lead.report.company.company_state',
+                'type'  => 'string',
+                'link'  => 'mautic_company_action',
+            ],
+            'comp.companycountry' => [
+                'label' => 'mautic.lead.report.company.company_country',
+                'type'  => 'string',
+                'link'  => 'mautic_company_action',
+            ],
+            'comp.companyindustry' => [
+                'label' => 'mautic.lead.report.company.company_industry',
+                'type'  => 'string',
+                'link'  => 'mautic_company_action',
+            ],
+            'companies_lead.is_primary' => [
+                'label' => 'mautic.lead.report.company.is_primary',
+                'type'  => 'bool',
+            ],
+        ];
+    }
+
+    /**
+     * @param Field[] $fields
+     * @param string  $prefix
+     *
+     * @return array
+     */
+    private function getFieldColumns($fields, $prefix)
+    {
+        $columns = [];
+        foreach ($fields as $f) {
+            switch ($f->getType()) {
+                case 'boolean':
+                    $type = 'bool';
+                    break;
+                case 'date':
+                    $type = 'date';
+                    break;
+                case 'datetime':
+                    $type = 'datetime';
+                    break;
+                case 'time':
+                    $type = 'time';
+                    break;
+                case 'url':
+                    $type = 'url';
+                    break;
+                case 'email':
+                    $type = 'email';
+                    break;
+                case 'number':
+                    $type = 'float';
+                    break;
+                default:
+                    $type = 'string';
+                    break;
+            }
+            $columns[$prefix.$f->getAlias()] = [
+                'label' => $f->getLabel(),
```

Comment:
Can we do something like https://github.com/mautic/mautic/blob/55db501b6e74857dc1aa33466811f36c10708269/app/bundles/ReportBundle/Event/ReportBuilderEvent.php#L251 for company columns so that each company label is prefixed with "Company". That way company fields aren't ambiguous. For example, "State." Is that contact state or company state?